rmf_traffic/src/rmf_traffic/Conflict.cpp

```diff
@@ -121,54 +121,116 @@ std::vector<ConflictData> DetectConflict::between(
-  const auto* t_bf = trajectory_b.finish_time();
+    if (D < 0)
+    {
+      // Do nothing
+    }
+    else if (D == 0)
+    {
+      double t = (-2 * coeffs[2]) / (6 * coeffs[3]);
+      double extrema = evaluate_spline(coeffs, t);
+      t_sols.emplace_back(extrema);
+    }
+    else
+    {
+      double t1 = ((-2 * coeffs[2]) + std::sqrt(D)) / (6 * coeffs[3]);
+      double t2 = ((-2 * coeffs[2]) - std::sqrt(D)) / (6 * coeffs[3]);
 
-  // Neither of these can be null, because both trajectories should have at
-  // least two elements.
-  assert(t_a0 != nullptr);
-  assert(t_bf != nullptr);
+      double extrema1 = evaluate_spline(coeffs, t1);
+      double extrema2 = evaluate_spline(coeffs, t2);
 
-  if(*t_bf < *t_a0)
-  {
-    // If Trajectory `b` finishes before Trajectory `a` starts, then there
-    // cannot be any conflict.
-    return false;
+      t_sols.emplace_back(extrema1);
+      t_sols.emplace_back(extrema2);
+    }
   }
 
-  const auto* t_b0 = trajectory_b.start_time();
-  const auto* t_af = trajectory_a.finish_time();
+  extrema[0] = *std::min_element(t_sols.begin(), t_sols.end());
+  extrema[1] = *std::max_element(t_sols.begin(), t_sols.end());
+}
 
-  // Neither of these can be null, because both trajectories should have at
-  // least two elements.
-  assert(t_b0 != nullptr);
-  assert(t_af != nullptr);
+BoundingBox get_bounding_box(const rmf_traffic::Spline& spline)
+{
+  BoundingBox bounding_box;
 
-  if(*t_af < *t_b0)
+  std::array<double, 2> extrema_x;
+  std::array<double, 2> extrema_y;
+
+  auto params = spline.get_params();
+
+  get_local_extrema(params.coeffs[0], extrema_x);
+  get_local_extrema(params.coeffs[1], extrema_y);
+
+  Eigen::Vector2d min_coord = Eigen::Vector2d{extrema_x[0], extrema_y[0]};
+  Eigen::Vector2d max_coord = Eigen::Vector2d{extrema_x[1], extrema_y[1]};
+
+  double char_length =  params.profile_ptr->get_shape()
+      ->get_characteristic_length();
+
+  if (char_length > 0)
   {
-    // If Trajectory `a` finished before Trajectory `b` starts, then there
-    // cannot be any conflict.
-    return false;
+    min_coord -= Eigen::Vector2d{char_length, char_length};
+    max_coord += Eigen::Vector2d{char_length, char_length};
   }
 
-  return true;
+  bounding_box.first = min_coord;
+  bounding_box.second = max_coord;
+
+  return bounding_box;
 }
 
-namespace {
+bool overlap(const BoundingBox& box_a, const BoundingBox& box_b)
+{
+  auto centre = [&](const BoundingBox& box) -> Eigen::Vector2d
+  {
+    return (0.5 * (box.second + box.first));
+  };
+
+  Eigen::Vector2d centre_a = centre(box_a);
+  Eigen::Vector2d centre_b = centre(box_b);
+  double width_a = std::abs(box_a.second[0] - box_a.first[0]);
+  double height_a = std::abs(box_a.second[1] - box_a.first[1]);
+  double width_b = std::abs(box_b.second[0] - box_b.first[0]);
+  double height_b = std::abs(box_b.second[1] - box_b.first[1]);
+
+  return std::abs(centre_b[0] - centre_a[0]) < 0.5 * (width_a + width_b)
```

Comment:
Maybe I'm missing something, but why not just compare the extrema? I.e.:

```
for (std::size_t i=0; i < 2; ++i)
{
  if (box_a.second[i] < box_b.first[i])
    return false;

  if (box_b.second[i] < box_a.first[i])
    return false;
}

return true;
```

Reply:
I did originally compare the extrema but in an inefficient (slower performance time) way as below
```
     return (box_a.second[0] > box_b.first[0] &&
    box_a.first[0] < box_b.second[0] &&
    box_a.second[1] > box_b.first[1] &&
    box_a.first[1] < box_b.second[1]);
```
You suggestion to return false as soon as a check fails is great and more performant. I have implemented this.